src/Runtime/Eval.hs

```diff
@@ -55,13 +56,16 @@ bind (BVal (Sig n _) defs _) = (n, do
       values <- mapM (eval . boardExpr) defs
       maybeBoard <- lookupName n
       case maybeBoard of
+         -- no board in the envs
          Nothing         -> return $ Vboard (fill (newBoard sz) sz defs values)
+         -- existing board in the env
          Just (Vboard b) -> return $ Vboard (fill b sz defs values)
-         _ -> error "TODO")
-   where
-      newBoard sz = array ((1,1), sz)
-                    (zip [(x,y) | x <- [1..(fst sz)], y <- [1..(snd sz)]] (repeat (Vs "?")))
```

Comment:
This PR is not the place to change this, but I realize now that, since we reject incomplete board definitions, placing undefined pieces is no longer necessary. This line can be changed, if for no other reason than that it may be confusing to someone down the road.